rpcserver.go

```diff
@@ -637,6 +637,73 @@ func createVinList(mtx *wire.MsgTx) []btcjson.Vin {
 	return vinList
 }
 
+// createVinList returns a slice of JSON objects for the inputs of the passed
+// transaction.
+func createVinListPrevOut(s *rpcServer, mtx *wire.MsgTx, chainParams *chaincfg.Params, vinExtra int) []btcjson.VinPrevOut {
+	vinList := make([]btcjson.VinPrevOut, len(mtx.TxIn))
+
+	// we don't call fetchInputTransactions() here because it may not be
+	// necessary so we defer until it is.  ( lazy load )
+	var txStore *blockchain.TxStore
```

Comment:
Since `blockchain.TxStore` is already a reference type, there is no need to make this a pointer.  Also, why not just do the input fetch here when `vinExtra` is set rather than checking on every loop iteration?  Something like:

```Go
var txStore blockchain.TxStore
if vinExtra == 1 {
	tx := btcutil.NewTx(mtx)
	txStoreNew, err := s.server.txMemPool.fetchInputTransactions(tx, true)
	if err == nil {
		txStore = txStoreNew
	}
}
```